validator.go

```diff
@@ -183,3 +191,25 @@ func WithStandardConstraintInterceptor(interceptor StandardConstraintInterceptor
 		c.resolver = interceptor(c.resolver)
 	}
 }
+
+// WithExtensionTypeResolver specifies a resolver to use when reparsing unknown
+// extension types. When dealing with dynamic file descriptor sets, passing this
+// option will allow extensions to be resolved using a custom resolver.
+func WithExtensionTypeResolver(extensionTypeResolver protoregistry.ExtensionTypeResolver) ValidatorOption {
+	return func(c *config) {
+		c.extensionTypeResolver = extensionTypeResolver
+	}
+}
+
+// WithAllowUnknownFields specifies if the presence of unknown field constraints
+// should cause compilation to fail with an error. When set to false, an unknown
+// field will simply be ignored, which will cause constraints to silently not be
+// applied. This condition may occur if a predefined constraint definition isn't
+// present in the extension type resolver, or when passing dynamic messages with
+// standard constraints defined in a newer version of protovalidate. The default
+// value is false, to prevent silently-incorrect validation from occurring.
```

Comment:
I'm a little worried by the `false` default value. I think this means every time a new user-defined rule is added as an extension and added to some field, the side of transportation that validates the payload would need to update the generated Go code, before the other side does. But I could have confused myself there and perhaps this is only a problem with dynamic messages (also not sure what I said made much sense)

Reply:
This should only impact dynamic file descriptor sets. Field constraints are specified in field options, and field options are stored in the file descriptor set, not on the protobufs themselves, so it only has to be consistent with itself. This is basically only a problem with dynamically-loaded file descriptor sets.

The reason for defaulting this to `false`:
- It won't break any _existing_ users, since the release that contains this will have a superset of all standard field constraints so far.
- If we default it to `true`, we can't change it to `false` without potentially breaking users later, which I think means it's now or never.
- Crucially, defaulting to `false` means that the behavior will just be incorrect. It will indeed pass validation, without running the custom predefined constraints at all.

Reply:
> field options are stored in the file descriptor set, not on the protobufs themselves

Yea that's where I confused myself, that was a rookie mistake on my part.

I agree with the reasoning.